### ISSUE TITLE
Style/IdenticalConditionalBranches cop error

### ISSUE THREAD

**Issue by akingo55** [rubocop/rubocop#12569]
Hi there,
I encountered `Style/IdenticalConditionalBranches` cop error.

--------

## Expected behavior

Until rubocop 1.56.4, I didn't encounter `Style/IdenticalConditionalBranches` cop error.

```bash
❯ bundle exec rubocop -d app/models/model_a.rb
For $HOME/repo: configuration from $HOME/repo/.rubocop.yml
configuration from $HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-rails-2.23.0/config/default.yml
configuration from $HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-rails-2.23.0/config/default.yml
Default configuration from $HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.56.4/config/default.yml
Inheriting configuration from $HOME/repo/.rubocop_todo.yml
Inheriting configuration from $HOME/repo/.rubocop_common.yml
Inheriting configuration from $HOME/repo/.rubocop_rails.yml
.rubocop.yml: Lint/MissingSuper:Exclude overrides the same parameter in .rubocop_todo.yml
.rubocop.yml: Metrics/BlockLength:Exclude overrides the same parameter in .rubocop_todo.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning $HOME/repo/app/models/model_a.rb
C

Offenses:

app/models/model_a.rb:67:7: C: [Correctable] Style/IdenticalConditionalBranches: Move self.foo ||= 0 out of the conditional.
      self.foo ||= 0
      ^^^^^^^^^^^^^^^^^^
app/models/model_a.rb:71:7: C: [Correctable] Style/IdenticalConditionalBranches: Move self.foo ||= 0 out of the conditional.
      self.foo ||= 0
      ^^^^^^^^^^^^^^^^^^

1 file inspected, 2 offenses detected, 2 offenses autocorrectable
```

## Actual behavior
After rubocop 1.57.0, I encountered the below error. (1.59.0 is also the same.)
```bash

❯ bundle exec rubocop -d app/models/model_a.rb
For $HOME/repo: configuration from $HOME/repo/.rubocop.yml
configuration from $HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-rails-2.23.0/config/default.yml
configuration from $HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-rails-2.23.0/config/default.yml
Default configuration from $HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/config/default.yml
Inheriting configuration from $HOME/repo/.rubocop_todo.yml
Inheriting configuration from $HOME/repo/.rubocop_common.yml
Inheriting configuration from $HOME/repo/.rubocop_rails.yml
.rubocop.yml: Lint/MissingSuper:Exclude overrides the same parameter in .rubocop_todo.yml
.rubocop.yml: Metrics/BlockLength:Exclude overrides the same parameter in .rubocop_todo.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning $HOME/repo/app/models/model_a.rb
An error occurred while Style/IdenticalConditionalBranches cop was inspecting $HOME/repo/app/models/model_a.rb:66:4.
undefined method `name' for s(:send,
  s(:self), :foo):RuboCop::AST::SendNode
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.30.0/lib/rubocop/ast/node/op_asgn_node.rb:18:in `name'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cop/style/identical_conditional_branches.rb:161:in `check_branches'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cop/style/identical_conditional_branches.rb:120:in `on_if'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:107:in `public_send'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:107:in `block (2 levels) in trigger_responding_cops'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:171:in `with_cop_error_handling'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:106:in `block in trigger_responding_cops'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:105:in `each'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:105:in `trigger_responding_cops'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:69:in `on_if'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.30.0/lib/rubocop/ast/traversal.rb:158:in `on_block'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.30.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.30.0/lib/rubocop/ast/traversal.rb:137:in `each'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.30.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.30.0/lib/rubocop/ast/traversal.rb:154:in `on_class'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:71:in `on_class'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.30.0/lib/rubocop/ast/traversal.rb:20:in `walk'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:87:in `investigate'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cop/team.rb:156:in `investigate_partial'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cop/team.rb:98:in `investigate'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:345:in `block in inspect_file'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:344:in `each'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:344:in `flat_map'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:344:in `inspect_file'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:287:in `block in do_inspection_loop'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:321:in `block in iterate_until_no_changes'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:314:in `loop'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:314:in `iterate_until_no_changes'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:283:in `do_inspection_loop'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:164:in `block in file_offenses'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:189:in `file_offense_cache'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:163:in `file_offenses'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:154:in `process_file'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:135:in `block in each_inspected_file'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:134:in `each'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:134:in `reduce'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:134:in `each_inspected_file'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:120:in `inspect_files'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/runner.rb:73:in `run'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cli/command.rb:11:in `run'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cli/environment.rb:18:in `run'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cli.rb:118:in `run_command'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cli.rb:125:in `execute_runners'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cli.rb:51:in `block in run'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cli.rb:77:in `profile_if_needed'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/lib/rubocop/cli.rb:43:in `run'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/exe/rubocop:19:in `block in <top (required)>'
$HOME/.rbenv/versions/3.0.6/lib/ruby/3.0.0/benchmark.rb:308:in `realtime'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rubocop-1.59.0/exe/rubocop:19:in `<top (required)>'
$HOME/.rbenv/versions/3.0.6/bin/rubocop:25:in `load'
$HOME/.rbenv/versions/3.0.6/bin/rubocop:25:in `<top (required)>'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/bundler-2.4.7/lib/bundler/cli/exec.rb:58:in `load'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/bundler-2.4.7/lib/bundler/cli/exec.rb:58:in `kernel_load'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/bundler-2.4.7/lib/bundler/cli/exec.rb:23:in `run'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/bundler-2.4.7/lib/bundler/cli.rb:492:in `exec'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/bundler-2.4.7/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/bundler-2.4.7/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/bundler-2.4.7/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/bundler-2.4.7/lib/bundler/cli.rb:34:in `dispatch'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/bundler-2.4.7/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/bundler-2.4.7/lib/bundler/cli.rb:28:in `start'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/bundler-2.4.7/exe/bundle:45:in `block in <top (required)>'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/bundler-2.4.7/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
$HOME/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/bundler-2.4.7/exe/bundle:33:in `<top (required)>'
$HOME/.rbenv/versions/3.0.6/bin/bundle:23:in `load'
$HOME/.rbenv/versions/3.0.6/bin/bundle:23:in `<main>'
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Style/IdenticalConditionalBranches cop was inspecting $HOME/repo/app/models/model_a.rb:66:4.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.59.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.0.6) [arm64-darwin21]
Finished in 0.7970109996385872 seconds
```

## Steps to reproduce the problem

In the case of the below codes on a model file, the error occurs.

```ruby
# model file
  before_create do
    if prepared?
      self.foo ||= 0
      self.bar ||= 'A'
    else
      self.foo ||= 0
      self.bar ||= 'B'
    end
  end
```

## RuboCop version

```
❯ bundle exec rubocop -V
1.59.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.0.6) [arm64-darwin21]
  - rubocop-rails 2.23.0
```
